c2corg_ui/static/js/map/map.js

```diff
@@ -429,13 +443,43 @@ app.MapController.prototype.handleDraw_ = function(event) {
  * @private
  */
 app.MapController.prototype.handleModify_ = function(event) {
-  // TODO handle lines as well, not only points
-  if (this.drawType != 'Point') {
-    alert('Feature type not supported for editing');
-    return;
-  }
-  var feature = event.features.item(0);
-  this.scope_.$root.$emit('mapFeatureChange', feature);
+  var features = event.features.getArray();
+  this.scope_.$root.$emit('mapFeaturesChange', features);
+};
+
+
+/**
+ * @private
+ */
+app.MapController.prototype.addTrackImporter_ = function() {
+  var dragAndDropInteraction = new ol.interaction.DragAndDrop({
+    formatConstructors: [
+      ol.format.GPX
+    ]
+  });
+  dragAndDropInteraction.on('addfeatures', function(event) {
+    var features = event.features;
+    if (features.length) {
+      features.forEach(this.simplifyFeature_);
+      this.showFeatures_(features);
+      this.scope_.$root.$emit('mapFeaturesChange', features);
+    }
+  }.bind(this));
+  this.map.addInteraction(dragAndDropInteraction);
+};
+
+
+/**
+ * @param {ol.Feature} feature Feature to process.
+ * @return {ol.Feature}
+ * @private
+ */
+app.MapController.prototype.simplifyFeature_ = function(feature) {
+  var geometry = feature.getGeometry();
+  // simplify geometry with a tolerance of 20 meters
+  geometry = geometry.simplify(20);
```

Comment:
10m?

My GPS is set to save points every 50m and it's already quite enough since there are often errors and poor precision => it's not that important to be really precise.
See for instance one of my outings: http://www.camptocamp.org/outings/739447/fr/la-grande-lui-en-boucle-par-les-5-cols#map_container